### PR TITLE
Add links to What's next section of Annotations page

### LINKED ISSUE #43086: Signpost from "Annotations" to "Well-Known Labels, Annotations and Taints" reference page
This page should probably link to resources for defining standard annotations. Especially the thinks mentioned as examples. That would unify it a bit (for example contact address). Maybe link to https://kubernetes.io/docs/reference/labels-annotations-taints/?

### PR DESCRIPTION
**Pull request by steve-hardman**
Added relevant links in 'What Section' of Annotations page.

Fixes #43086

### CODE REVIEW

**Review by ecki — APPROVED** [review]
Looks good to me, (I think there is no other offices places listing some best practice well known annotations, yet).

Not an official reviewer, but here is my +1 since I raised the issue

nit …‘What’s Next’ section… in commit message

File: content/en/docs/concepts/overview/working-with-objects/annotations.md [hunk 1]
```diff
@@ -93,4 +93,5 @@ spec:
 
 ## {{% heading "whatsnext" %}}
 
-Learn more about [Labels and Selectors](/docs/concepts/overview/working-with-objects/labels/).
+- Learn more about [Labels and Selectors](/docs/concepts/overview/working-with-objects/labels/).
+- [Well-known labels, Annotations and Taints](/docs/reference/labels-annotations-taints/)
```
Comment (dipesh-rawat) [inline content/en/docs/concepts/overview/working-with-objects/annotations.md:97]: Changes looks good.

_nit: Refer suggestion_
```suggestion
- Find [Well-known labels, Annotations and Taints](/docs/reference/labels-annotations-taints/)
```

**Comment by steve-hardman** [comment]
@dipesh-rawat I have made the changes you asked plus the build is also passed now; can you look again please.

**Review by dipesh-rawat — COMMENTED** [review]
Thank you for your contribution!
/lgtm

Deploy Preview: [deploy-preview-43087-annotations/#what-s-next](https://deploy-preview-43087--kubernetes-io-main-staging.netlify.app/docs/concepts/overview/working-with-objects/annotations/#what-s-next)

File: content/en/docs/concepts/overview/working-with-objects/annotations.md [hunk 2]
```diff
@@ -93,4 +93,5 @@ spec:
 
 ## {{% heading "whatsnext" %}}
 
-Learn more about [Labels and Selectors](/docs/concepts/overview/working-with-objects/labels/).
+- Learn more about [Labels and Selectors](/docs/concepts/overview/working-with-objects/labels/).
+- Find [Well-known labels, Annotations and Taints](/docs/reference/labels-annotations-taints/)
```
Comment (reylejano) [inline content/en/docs/concepts/overview/working-with-objects/annotations.md:97]: nit: add period
```suggestion
- Find [Well-known labels, Annotations and Taints](/docs/reference/labels-annotations-taints/).
```

**Pull request merged**